source/components/GlobalNavigation/components/GlobalNavigationNotifications/CardLink.js

```diff
@@ -0,0 +1,47 @@
+import React, { useCallback, useContext } from 'react';
+import PropTypes from 'prop-types';
+
+import NotificationsContext from '../../utils/NotificationContext';
+
+const CardLink = ({ track, model, children }) => {
+    const { markAsRead, goToDestination } = useContext(NotificationsContext);
+    const { type, isUnread, uri, latestEventUri } = model;
+    const onClick = useCallback((event) => {
+        event.preventDefault();
+
+        track({
+            action: 'click',
+            category: 'on-site-notifications',
+            label: type,
+            value: isUnread ? 1 : 0,
+        });
+
+        if (isUnread) {
+            markAsRead(uri, true)
```

Comment:
I don't think we need to mark the uri as read before going to it. The discussion service should handle that. See [here](https://github.com/Wikia/pandora/blob/master/service/discussion/src/main/java/com/wikia/discussionservice/services/AggregationService.java#L272). Any time a user visits a thread we emit a contentvisited event which is consumed by on-site notifications which then marks that notification as read. The markAsRead API is used when we want to mark the content as read _without_ the user visiting it.

That being said, it looks like announcements was added to ONS, and I'm not sure how that works. Maybe it's required b/c of that?

Reply:
tbh, i don't know. here's how it's defined in emberish DS https://github.com/Wikia/design-system/blob/IW-1022/app/components/on-site-notifications/notification-card.js#L110-L130

I've tried to recreate it 1:1 in terms of behaviour, so I didn't think about. maybe it can be ditched, but definitely not now 🙈

Reply:
👍